### PR TITLE
Cleaned up the developer guide.

### PR DESCRIPTION
**Pull request by dblock**
### Description

Took me some time to figure out how to run tests. 
A small cleanup of the dev guide. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -1,43 +1,70 @@
...
+If your OpenSearch docker instance is running, you can execute the test suite directly.
 
-# Run the test suite
+```
 $ nox -rs test
 ```
 
-## Integration Tests
-To run the integration tests locally, run:
+To run tests against different versions of OpenSearch, use `run-tests [with/without security] [version]`:
```
Comment (VachaShah) [inline DEVELOPER_GUIDE.md:40]: Can we also add that tests can be run against unreleased OpenSearch as well using `./.ci/run-tests opensearch false SNAPSHOT`. To do so: OpenSearch instance can be started by building it from the OpenSearch repo branch. Related workflow: https://github.com/opensearch-project/opensearch-py/blob/main/.github/workflows/integration-unreleased.yml

Comment (dblock) [reply DEVELOPER_GUIDE.md:40]: Done.

**Comment by VachaShah** [comment]
The macos tests seem to be failing, @dblock any insights?

**Review by VachaShah — APPROVED** [review]
LGTM!

**Pull request merged**